tests/main_test.go

```diff
@@ -0,0 +1,171 @@
+//go:build integration
+
+// Copyright (c) 2024 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package tests
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/canonical/pebble/internals/servicelog"
+)
+
+// TestMain builds the pebble binary before running the integration tests.
+func TestMain(m *testing.M) {
+	goBuild := exec.Command("go", "build", "-o", "../pebble", "../cmd/pebble")
```

Comment:
Not a fan of invoking go build in a test, but if that's the path @benhoyt is happy with, then I am ok.

But we should be able to run the integration tests against any pre-compiled pebble binary. This way the juju team can add these tests into our gating test pipeline.

So I suggest we add a flag to the test executable like `--no-build`, where instead it uses `exec.LookPath`.

Reply:
Yeah, I really like that it makes them just work by default. However, it's a good idea to allow a more explicit override. I'd prefer a slightly more explicit approach: a flag `-pebblebin` that, if set, uses that as the Pebble binary rather than building one.

Reply:
That way you can point it at a binary you or the Juju CI or whatever has just built, rather than necessarily having to install it or put it in PATH.